Sources/Build/BuildPlan.swift

```diff
@@ -562,6 +566,21 @@ public final class SwiftTargetBuildDescription {
         self.fs = fs
         self.tempsPath = buildParameters.buildPath.appending(component: target.c99name + ".build")
         self.derivedSources = Sources(paths: [], root: tempsPath.appending(component: "DerivedSources"))
+        self.extensionEvaluationResults = extensionEvaluationResults
+
+        // Add any derived source paths declared by build-tool extensions that were applied to this target.  We do
+        // this here and not just in the LLBuildManifestBuilder because we need to include them in any situation
+        // where sources are processed, e.g. when determining names of object files, etc.
+        for command in extensionEvaluationResults.reduce([], { $0 + $1.commands }) {
+            // Prebuild and postbuild commands are handled outside the build system.
+            if case .buildToolCommand(_, _, _, _, _, _, _, let derivedSourcePaths) = command {
+                // TODO: What should we do if we find non-Swift sources here?
```

Comment:
There are still several diagnostics to add of this nature — it should be an error, earlier on, if an extension emits source code that isn't compatible with the target into which it's being compiled.  This is because SwiftPM doesn't yet allow mixed sources, and it's beyond the scope of the extensibility proposal to change that.  In this case, this error should be caught and reported earlier on.